File: week-5/getSquares.js
var numArrayOne = [1, 4, 3, 6, 16, 5, 24]
var newArr = []

const getSquares = (numArr) => {
    numArrayOne.forEach(checkSquare)
    newArr.sort
    return newArr
}

const checkSquare = (n) => {
    var sqrt = Math.sqrt(n)
    if (/\.0/.test(sqrt)) {  // don't do regex on floating points
        newArr.push(n)
    }
}

console.log(getSquares(numArrayOne))

File: week-5/get_squares.rb
numArrOne = [4, 5, 6, 16, 15, 4]

def get_squares numArr
    arr = []
    numArr.each do |n|
        sqrt = Math.sqrt(n)
        if sqrt % 1 == 0 #sqrt.to_s =~ /\.0/
            arr.push(n)
        end 
    end 
    arr.sort.uniq
    # arr - no need for a return line
end 

puts get_squares numArrOne

# puts Math.sqrt(35)
